File: StudentGrade.java
import java.util.Scanner;
import java.util.Scanner;

public class StudentGrade {
    public static void main(String[] args) {


	System.out.println ("Welcome to semi colon Africa StudentGrade System");

        Scanner input = new Scanner(System.in);

        System.out.print("what are the numbers of students: ");
        int numofStudents = input.nextInt();

        System.out.print("what are the numbers of subjects offerred: ");
        int numofSubjects = input.nextInt();

        int[][] scores = new int[numofStudents][numofSubjects];
        int[] studentTotals = new int[numofStudents];
        double[] studentAverages = new double[numofStudents];

        for (int studentIndex = 0; studentIndex < numofStudents; studentIndex++) {
            System.out.println("collecting scores of students " + (studentIndex + 1));
            int totalScore = 0;

            for (int subjectIndex = 0; subjectIndex < numofSubjects; subjectIndex++) {
                int score = -1;

                for (int attempt = 0; score < 0 || score > 100; attempt++) {
                   System.out.print("Please enter the score for Subjects " + (subjectIndex + 1) + ": ");

                    score = input.nextInt();

                    if (score < 0 || score > 100) {
                        System.out.println("invalid score, Student score can only be between 0 and 100.");
                    }
                }

                scores[studentIndex][subjectIndex] = score;
                totalScore += score;
            }

            studentTotals[studentIndex] = totalScore;
            studentAverages[studentIndex] = (double) totalScore / numofSubjects;
        }

        System.out.println("Students Grade Report");

        int classTotal = 0;
        int lowestScore = studentTotals[0];
        int highestScore = studentTotals[0];
        int leastStudent = 0;
        int topStudent = 0;

        for (int studentIndex = 0; studentIndex < numofStudents; studentIndex++) {
            int total = studentTotals[studentIndex];
            double average = studentAverages[studentIndex];
            System.out.printf("Student %d Total: %d, Average: %.2f%n", studentIndex + 1, total, average);
	    


            classTotal += studentTotals[studentIndex];

            if (studentTotals[studentIndex] > highestScore) {
                highestScore = studentTotals[studentIndex];
                topStudent = studentIndex;
            }

            if (studentTotals[studentIndex] < lowestScore) {
                lowestScore = studentTotals[studentIndex];
                leastStudent = studentIndex;
            }
        }

        double classAverage = (double) classTotal / (numofStudents * numofSubjects);

        System.out.println("The Class Average is : " + String.format("%.2f", classAverage));

        System.out.println("Best Performing Student is : Student " + (topStudent + 1)
                + " (Total: " + highestScore + ")");
        System.out.println("Lowest Performing Student is: Student " + (leastStudent + 1)
                + " (Total: " + lowestScore + ")");
    }
}

	 

	  
	

File: JAVASCRIPT/StudentGrade.js
console.log("Welcome to semi colon Africa StudentGrade System");

const prompt = require('prompt-sync')();

let numofStudents = parseInt(prompt("what are the numbers of students: "));
let numofSubjects = parseInt(prompt("what are the numbers of subjects offerred: "));

let scores = [];
let studentTotals = new Array(numofStudents).fill(0);
let studentAverages = new Array(numofStudents).fill(0);

for (let studentIndex = 0; studentIndex < numofStudents; studentIndex++) {
    console.log("collecting scores of students " + (studentIndex + 1));
    let totalScore = 0;
    scores[studentIndex] = [];

    for (let subjectIndex = 0; subjectIndex < numofSubjects; subjectIndex++) {
        let score = -1;

        while (score < 0 || score > 100) {
            score = parseInt(prompt("Please enter the score for Subjects " + (subjectIndex + 1) + ": "));
            if (score < 0 || score > 100 || isNaN(score)) {
                console.log("invalid score, Student score can only be between 0 and 100.");
            }
        }

        scores[studentIndex][subjectIndex] = score;
        totalScore += score;
    }

    studentTotals[studentIndex] = totalScore;
    studentAverages[studentIndex] = totalScore / numofSubjects;
}

console.log("Students Grade Report");

let classTotal = 0;
let lowestScore = studentTotals[0];
let highestScore = studentTotals[0];
let leastStudent = 0;
let topStudent = 0;

for (let studentIndex = 0; studentIndex < numofStudents; studentIndex++) {
    let total = studentTotals[studentIndex];
    let average = studentAverages[studentIndex];
    console.log("Student " + (studentIndex + 1) + " Total: " + total + ", Average: " + average.toFixed(2));

    classTotal += total;

    if (total > highestScore) {
        highestScore = total;
        topStudent = studentIndex;
    }

    if (total < lowestScore) {
        lowestScore = total;
        leastStudent = studentIndex;
    }
}

let classAverage = classTotal / (numofStudents * numofSubjects);

console.log("The Class Average is : " + classAverage.toFixed(2));
console.log("Best Performing Student is : Student " + (topStudent + 1) + " (Total: " + highestScore + ")");
console.log("Lowest Performing Student is: Student " + (leastStudent + 1) + " (Total: " + lowestScore + ")");
